server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -657,12 +659,28 @@ public void applyClusterState(ClusterChangedEvent event) {
                 }
                 removeFinishedSnapshotFromClusterState(event);
                 finalizeSnapshotDeletionFromPreviousMaster(event);
+                // TODO org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testDeleteOrphanSnapshot fails right after election here
+                assert event.previousState().nodes().isLocalNodeElectedMaster() != false || assertConsistency(event.state());
```

Comment:
`X != false <==> X`

Reply:
Note that we'll have a similar problem when transitioning from a master that did not keep the routing table in sync to a master that now does. We will probably have to schedule a clean-up task that brings these two in sync again. There's no guarantee that that task will run before any other tasks, so the assertion might not only be violated while `event.previousState().nodes().isLocalNodeElectedMaster() == false`, but also on some follow-up cluster state changes. I'm wondering if we need to introduce some state to SnapshotsService to say whether we've properly cleaned up as a master and then make that assertion dependent on that boolean variable. Something we can look into in a follow-up.